packages/node/src/builder.ts

```diff
@@ -139,47 +139,40 @@ export class Builder {
 
     const done = logger.time.info('Extracted in')
 
-    // limit concurrency since we might parse a lot of files
-    const promises = files.map((file) => limit(() => this.extractFile(ctx, file)))
-    await Promise.allSettled(promises)
+    files.map((file) => this.extractFile(ctx, file))
 
     done()
   }
 
-  toString = () => {
-    const ctx = this.getContextOrThrow()
-    return ctx.getCss()
-  }
-
   isValidRoot = (root: Root) => {
     const ctx = this.getContextOrThrow()
     let valid = false
 
     root.walkAtRules('layer', (rule) => {
-      if (ctx.layers.isValidParams(rule.params)) {
+      if (ctx.isValidLayerParams(rule.params)) {
```

Comment:
I think keeping the `isValidParams` in the layer class is still a good idea. Don't think we should move it to the root, except I'm missing some context.